services/plansProposalsFacade/plansProposalsFacade-portlet/src/test/java/com/ext/portlet/service/impl/XCoLabTest.java

```diff
@@ -169,118 +197,213 @@ protected void deleteUsers() throws SystemException, PortalException {
+            User user = ((PasswordPolicyLocalServiceBaseImpl) passwordPolicyLocalService).getUserPersistence()
+                    .findByPrimaryKey(10115L);
+            if (user != null) {
+                PasswordPolicy passwordPolicy = passwordPolicyLocalService
+                        .getDefaultPasswordPolicy(user.getCompanyId());
+                if (passwordPolicy == null) {
+                    passwordPolicyLocalService.addPasswordPolicy(user.getCompanyId(), true, "Default Password Policy",
+                            "Default Password Policy", true, false, 0L, false, true, 6, 0, 6, 0, 0, 0, "", false, 0,
+                            false, 8640000L, 0L, 0, false, 3, 600L, 0L, 0L, new ServiceContext());
+                }
+            }
+        }
+        // plan section for subproposal list
+        int planSectionDefinitionId = 1300907;
+        PlanSectionDefinition psd = null;
+        try {
+            psd = planSectionDefinitionLocalService.getPlanSectionDefinition(planSectionDefinitionId);
+        } catch (NoSuchPlanSectionDefinitionException e) {
+            psd = planSectionDefinitionLocalService.createPlanSectionDefinition(planSectionDefinitionId);
+            psd.setType("PROPOSAL_LIST_TEXT_REFERENCE");
+            psd.setAdminTitle("Sub-proposals Global");
+            psd.setTitle("Sub-proposals");
+            psd.setCharacterLimit(7000);
+            psd.setFocusAreaId(0);
+            psd.setLocked(false);
+            planSectionDefinitionLocalService.addPlanSectionDefinition(psd);
+        }
+                
         createPointTypes();
         createContestPhaseTypes();
+        createContest();
     }
 
-    protected static void createContestPhaseTypes() throws SystemException {
-        ContestPhaseType cpt = contestPhaseTypeLocalService.createContestPhaseType(1);
-        cpt.setName("Proposal creation");
-        cpt.setStatus("OPEN_FOR_SUBMISSION");
-        contestPhaseTypeLocalService.updateContestPhaseType(cpt);
-
-        cpt = contestPhaseTypeLocalService.createContestPhaseType(15);
-        cpt.setName("Winners selection");
-        cpt.setStatus("VOTING");
-        contestPhaseTypeLocalService.updateContestPhaseType(cpt);
-
-        cpt = contestPhaseTypeLocalService.createContestPhaseType(16);
-        cpt.setName("Semi-Finalist selection");
-        cpt.setStatus("CLOSED");
-        contestPhaseTypeLocalService.updateContestPhaseType(cpt);
-
-        cpt = contestPhaseTypeLocalService.createContestPhaseType(17);
-        cpt.setName("Winners awarded");
-        cpt.setStatus("CLOSED");
-        contestPhaseTypeLocalService.updateContestPhaseType(cpt);
-
-        cpt = contestPhaseTypeLocalService.createContestPhaseType(18);
-        cpt.setName("Proposal revisions");
-        cpt.setStatus("OPEN_FOR_EDIT");
-        contestPhaseTypeLocalService.updateContestPhaseType(cpt);
+    private static void createContest() throws SystemException, PortalException {
```

Comment:
New method to create a contest as some tests within this module were requiring such contest record.